fs/userfs/fs_userfs.c

```diff
@@ -899,6 +904,85 @@ static int userfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
   return resp->ret;
 }
 
+/****************************************************************************
+ * Name: userfs_fchstat
+ *
+ * Description:
+ *   Change information about an open file associated with the file
+ *   descriptor 'fd'.
+ *
+ ****************************************************************************/
+
+static int userfs_fchstat(FAR const struct file *filep,
+                          FAR const struct stat *buf, int flags)
+{
+  FAR struct userfs_state_s *priv;
+  FAR struct userfs_fchstat_request_s *req;
+  FAR struct userfs_fchstat_response_s *resp;
+  ssize_t nsent;
+  ssize_t nrecvd;
+  int ret;
+
+  DEBUGASSERT(filep != NULL &&
+              filep->f_inode != NULL &&
+              filep->f_inode->i_private != NULL);
+  priv = filep->f_inode->i_private;
+
+  /* Get exclusive access */
+
+  ret = nxsem_wait(&priv->exclsem);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  /* Construct and send the request to the server */
+
+  req           = (FAR struct userfs_fchstat_request_s *)priv->iobuffer;
+  req->req      = USERFS_REQ_FCHSTAT;
+  req->openinfo = filep->f_priv;
+  req->buf      = *buf;
+  req->flags    = flags;
+
+  nsent = psock_sendto(&priv->psock, priv->iobuffer,
+                       sizeof(struct userfs_fchstat_request_s), 0,
+                       (FAR struct sockaddr *)&priv->server,
+                       sizeof(struct sockaddr_in));
+  if (nsent < 0)
+    {
+      ferr("ERROR: psock_sendto failed: %d\n", (int)nsent);
+      nxsem_post(&priv->exclsem);
+      return (int)nsent;
+    }
+
+  /* Then get the response from the server */
+
+  nrecvd = psock_recvfrom(&priv->psock, priv->iobuffer, IOBUFFER_SIZE(priv),
+                          0, NULL, NULL);
+  nxsem_post(&priv->exclsem);
+
+  if (nrecvd < 0)
+    {
+      ferr("ERROR: psock_recvfrom failed: %d\n", (int)nrecvd);
+      return (int)nrecvd;
+    }
+
+  if (nrecvd != sizeof(struct userfs_fchstat_response_s))
+    {
+      ferr("ERROR: Response size incorrect: %u\n", (unsigned int)nrecvd);
```

Comment:
```suggestion
      ferr("ERROR: Response size incorrect: %zd\n", nrecvd);
```
I think there is no point printing it as unsigned, right?

Reply:
Done.